src/enzo/ActiveParticle_SmartStar.h

```diff
@@ -0,0 +1,450 @@
+    if (groupsize[i] != 1) {
+      for (j=1; j<groupsize[i]; j++) {
+    MergedParticles[i]->SmartMerge(
+          static_cast<active_particle_class*>(ParticleList[grouplist[i][j]]));
+        if (ParticleList[grouplist[i][j]]->DisableParticle(
+                LevelArray, 
+                MergedParticles[i]->ReturnCurrentGrid()->ReturnProcessorNumber()
+              ) == FAIL)
+        {
+          ENZO_FAIL("MergeSmartStars: DisableParticle failed!\n");
+        }
+      }
+    }
+  }
+  
+  ParticleList.clear();
+
+  free(groupsize);
+  groupsize = NULL;
+  for (i=0; i<*ngroups; i++)
+    free(grouplist[i]);
+  free(grouplist);
+  grouplist = NULL;
+  
+  /* Loop over the grids and check if any of the merged particles have
+     moved. If so, disable the particle on the current grid and assign
+     it to the new grid*/
+
+  int NewGrid = -1;
+
+  for (i = 0; i < *ngroups; i++) {
+    if (MergedParticles[i]->ReturnCurrentGrid()->PointInGrid(
+            MergedParticles[i]->ReturnPosition()) == false) {
+      // Find the grid to transfer to 
+      for (j = 0; j < NumberOfGrids; j++) {
+        if (LevelGrids[j]->GridData->PointInGrid(
+                MergedParticles[i]->ReturnPosition())) {
+          NewGrid = j;
+          break;
+        }
+      }
+      if (NewGrid == -1)
+        ENZO_FAIL("Cannot assign particle to grid after merging!\n");
+      int OldProc = MergedParticles[i]->CurrentGrid->ReturnProcessorNumber();
+      active_particle_class *temp = static_cast<active_particle_class*>(
+          MergedParticles[i]->clone());
+      MergedParticles[i]->DisableParticle(LevelArray, 
+          LevelGrids[NewGrid]->GridData->ReturnProcessorNumber()); 
+      if (LevelGrids[NewGrid]->GridData->AddActiveParticle(
+              static_cast<ActiveParticleType*>(temp)) == FAIL)
+      	ENZO_FAIL("Active particle grid assignment failed!\n");
+      if (MyProcessorNumber == OldProc) {
+        MergedParticles.erase(i);
+        MergedParticles.insert(*temp);
+      }
+      else if (MyProcessorNumber != temp->CurrentGrid->ReturnProcessorNumber())
+        delete temp;
+      MergedParticles[i]->AssignCurrentGrid(LevelGrids[NewGrid]->GridData);
+    }
+  }
+
+  delete [] LevelGrids;
+
+  *nParticles = *ngroups;
+}
+
+
+int GenerateGridArray(LevelHierarchyEntry *LevelArray[], int level,
+		      HierarchyEntry **Grids[]);
+ 
+int AssignActiveParticlesToGrids(
+    ActiveParticleList<ActiveParticleType>& ParticleList, int nParticles, 
+    LevelHierarchyEntry *LevelArray[]); 
+
+template <class active_particle_class>
+int ActiveParticleType_SmartStar::AfterEvolveLevel(
+    HierarchyEntry *Grids[], TopGridData *MetaData,
+    int NumberOfGrids, LevelHierarchyEntry *LevelArray[], 
+    int ThisLevel, int TotalStarParticleCountPrevious[],
+    int SmartStarID)
+{
+
+  /* Accreting particles live on the maximum refinement level.  If we are on a lower level, this does not concern us */
+
```

Comment:
Why is 'Accreting particle' being used here?

Reply:
Typo. Fixed now.